src/modules/buyer/controller/buyerController.ts

```diff
@@ -0,0 +1,24 @@
+import { Response,Request } from "express";
+import productRepositories from "../../product/repositories/productRepositories";
+import httpStatus from "http-status";
+
+
+const viewSpecificProduct = async (req:Request,res:Response) => {
+try{
+
+    const {id} = req.params;
+    const singleProduct = await productRepositories.findProductById(id);
+    res.status(httpStatus.OK).json({
+        message: "Products is fetched successfully.",
+        singleProduct
```

Comment:
We are returning the data object `data` with the property `singleProduct`. Besides `singleProduct`, please rename it to `product`.